rajawali/src/main/java/c/org/rajawali3d/core/AFrameDelegate.java

```diff
@@ -0,0 +1,234 @@
+                if (callback.callFrameStart()) {
+                    frameStartCallbacks.add(callback);
+                }
+                if (callback.callFrameEnd()) {
+                    frameEndCallbacks.add(callback);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeFrameCallback(@NonNull final FrameCallback callback) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                if (callback.callFrameStart()) {
+                    frameStartCallbacks.remove(callback);
+                }
+                if (callback.callFrameEnd()) {
+                    frameEndCallbacks.remove(callback);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean clearFrameCallbacks() {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                frameStartCallbacks.clear();
+                frameEndCallbacks.clear();
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean addAnimation(final Animation anim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.add(anim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeAnimation(final Animation anim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.remove(anim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean replaceAnimation(final Animation oldAnim, final Animation newAnim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.set(animations.indexOf(oldAnim), newAnim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean addAnimations(final Collection<Animation> anims) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.addAll(anims);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean clearAnimations() {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.clear();
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    //
+    // ACoreComponent method overrides
+    //
+
+    @RenderThread
+    @Override
+    @CallSuper
+    public void onFrameStart(double deltaTime) throws InterruptedException {
+        if (!isEnabled) {
```

Comment:
Yeah, I could move the enable check to the `RenderControl` delegation loops before the call is made, like this method does for animations, or I could just leave this method abstract in `ACoreComponent` and call `runQueuedTasks()` directly from here and in RenderControl, or introduce a `doCoreComponentStuffForFrameStart()` method so if the logic changes in the future there is still only one place to change. 

Out of curiosity, what kinds of problems can come up by doing something before calling super methods?  You mentioned debugging, but I don't see the issues/

And you made me look up PMD. 👓  Seems like a super-lint.